pkgs/xdg-desktop-portal-cosmic/package.nix

```diff
@@ -47,12 +50,15 @@ rustPlatform.buildRustPackage rec {
       --replace-fail 'Exec=/bin/false' 'Exec=${lib.getExe' coreutils "true"}'
   '';
 
+  dontCargoInstall = true;
+
+  makeFlags = [
+    "CARGO_TARGET_DIR=target/${stdenv.hostPlatform.rust.cargoShortTarget}"
+    "prefix=${placeholder "out"}"
+  ];
+
   postInstall = ''
-    mkdir -p $out/share/{dbus-1/services,icons,xdg-desktop-portal/portals}
-    cp -r data/icons $out/share/icons/hicolor
-    cp data/*.service $out/share/dbus-1/services/
-    cp data/cosmic.portal $out/share/xdg-desktop-portal/portals/
-    cp data/cosmic-portals.conf $out/share/xdg-desktop-portal/
+    mv $out/libexec $out/bin
```

Comment:
I wonder if we should just leave it in `libexec` where upstream has it and just update `env.XDP_COSMIC` in `cosmic-session` derivation? But either way that can be out of scope for this PR

Reply:
I thought the same to myself, I can update it open another PR.